### ISSUE TITLE
Allow `assert` module to provide `$traceurRuntime.type.*` and `$traceurRuntime.genericType`

### ISSUE THREAD

**Issue by tbosch** [google/traceur-compiler#1706]
When transpiling AtScript to ES6, the only reference to `$traceurRuntime` are `$traceurRuntime.type.*` and `$traceurRuntime.genericType(...)`. However, those references are very closely related to the assertion module, so it would be great to have an option to use the implicitly imported `assert` module instead.

Traceur 0.0.82

**Comment by arv** [comment]
The goal is to make the assertion module non configurable. Things like type semantics is an important part of the goal, making that pluggable defeats that.

**Comment by tbosch** [comment]
True, but the code that is generated right now looks like [this](https://google.github.io/traceur-compiler/demo/repl.html#%2F%2F%20Options%3A%20--types%20--type-assertions%0Aclass%20Test%20%7B%0A%20%20m(a%3Astring)%20%7B%0A%20%20%7D%0A%7D):
```
assert.argumentTypes(a, $traceurRuntime.type.string);
```

So what a type really means is defined by the assertion module. Whether the token is stored inside of `$traceurRuntime` or `assert` does not change/control the semantics.

**Comment by arv** [comment]
For builtins I think we can change to strings. The generics is a bit worse.

**Comment by tbosch** [comment]
Agree, strings for builtins would be great. How about just using an object literal for generics?
E.g. 
```
{ type: List, args: ['string']}
```
